iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/posix_call.hpp

```diff
@@ -102,7 +102,13 @@ struct PosixCallDetails
 ///        // when your posix call signals failure by returning the errno value instead of setting the errno use
 ///        // .returnValueMatchesErrno() instead of .successReturnValue(_)
 /// @endcode
-#define posixCall(f) internal::createPosixCallBuilder(f, #f, __FILE__, __LINE__, __PRETTY_FUNCTION__)
+#define posixCall(f)                                                                                                   \
+    internal::createPosixCallBuilder(                                                                                  \
+        f,                                                                                                             \
+        #f,                                                                                                            \
+        __FILE__,                                                                                                      \
+        __LINE__,                                                                                                      \
+        __PRETTY_FUNCTION__) // NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay,hicpp-no-array-decay)
```

Comment:
Please explain why the warnings can be suppressed.

Reply:
done